internal/rbd/rbd_util.go

```diff
@@ -200,6 +200,20 @@ func createImage(ctx context.Context, pOpts *rbdVolume, cr *util.Credentials) er
 	return nil
 }
 
+// Open the rbdVolume after it has been connected.
+func (rv *rbdVolume) open() (*librbd.Image, error) {
+	ioctx, err := rv.conn.GetIoctx(rv.Pool)
+	if err != nil {
+		return nil, ErrImageNotFound{rv.Pool, err}
```

Comment:
this seems not correct right?, due to some internal error, we are returning ImageNotFound.

Reply:
Maybe, it should probably make a distinction between PoolNotFound and some PermissionError. The connection is already available, and that requires some authentication too. I'll update it to return different errors.